Adding I2S Support for the RW612

EmilioCBen opened the pull request:
Addresses https://github.com/zephyrproject-rtos/zephyr/issues/87239.

DerekSnell reviewed (dismissed):
Thanks @EmilioCBen 

You have compliance issues to clean up.  But the i2s_speed test works for me on both boards.  And your changes look good to me.  Thanks

decsny commented:
im not sure why did you split the board changes into two commit, neither of them work without the other, and both are just different parts of the code to enable i2s for the board, why not combine them to one commit

mmahadevan108 commented on boards/nxp/rd_rw612_bga/Kconfig:
```diff
@@ -0,0 +1,5 @@
+# Copyright 2024 NXP
```
2025. Also there is an overlap with this PR: https://github.com/zephyrproject-rtos/zephyr/pull/87594

EmilioCBen replied:
Thank you for making me aware of this PR I will stay in sync, also I will update the copyrights.

mmahadevan108 commented on boards/nxp/rd_rw612_bga/init.c:
```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2022, 2024 NXP
```
2024-25

EmilioCBen replied:
Updated

mmahadevan108 commented on boards/nxp/rd_rw612_bga/CMakeLists.txt:
```diff
@@ -14,4 +14,5 @@ if(CONFIG_NXP_RW6XX_BOOT_HEADER)
   zephyr_compile_definitions(BOARD_FLASH_SIZE=CONFIG_FLASH_SIZE*1024)
   zephyr_library()
   zephyr_library_sources(MX25U51245GZ4I00_FCB.c)
+  zephyr_library_sources(init.c)
```
This should be moved outside the CONFIG as we need this file even when we have a bootloader.

EmilioCBen replied:
Thank you again for pointing this out, updated.

EmilioCBen replied:
Thank you again for pointing this out, updated.

mmahadevan108 commented on boards/nxp/frdm_rw612/init.c:
```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2022, 2024 NXP
```
Same comments as rd_rw612_bga

EmilioCBen replied:
Updated.

EmilioCBen commented:
> im not sure why did you split the board changes into two commit, neither of them work without the other, and both are just different parts of the code to enable i2s for the board, why not combine them to one commit

Sorry, I thought this would be better for clarity and since the second commit is what enables the support on the board dts I thought I separate why the init.c was included with the dts node being added.

EmilioCBen commented:
> im not sure why did you split the board changes into two commit, neither of them work without the other, and both are just different parts of the code to enable i2s for the board, why not combine them to one commit

Sorry, I thought this would be better for clarity and since the second commit is what enables the support on the board dts I thought I separate why the init.c was included with the dts node being added.

decsny commented:
> > im not sure why did you split the board changes into two commit, neither of them work without the other, and both are just different parts of the code to enable i2s for the board, why not combine them to one commit
> 
> Sorry, I thought this would be better for clarity and since the second commit is what enables the support on the board dts I thought I separate why the init.c was included with the dts node being added.

it would make more sense to put the loop test code in init.c in the same commit as enabling the test, and put the clocks in the same commit as all the other normal board enablement code

EmilioCBen committed:
boards: nxp: Added I2S Support for RW612

Added I2S support for RW612.

Signed-off-by: Emilio Benavente <emilio.benavente@nxp.com>

EmilioCBen committed:
tests: drivers: i2s: Added I2S Testing for RW612

Added Internal pin connection for
I2S Loopback testing when I2S is enabled.
Added I2S testing support for RW612.

Signed-off-by: Emilio Benavente <emilio.benavente@nxp.com>

EmilioCBen commented:
> it would make more sense to put the loop test code in init.c in the same commit as enabling the test, and put the clocks in the same commit as all the other normal board enablement code

That does make more sense I will update the PR to reflect this.

mmahadevan108 commented on boards/nxp/frdm_rw612/CMakeLists.txt:
```diff
@@ -1,9 +1,11 @@
+
 if(CONFIG_NXP_RW6XX_BOOT_HEADER)
   zephyr_compile_definitions(BOARD_FLASH_SIZE=CONFIG_FLASH_SIZE*1024)
   zephyr_library()
```
This should move outside the if as well

EmilioCBen replied:
Sure that make sense, I can move this as well.

mmahadevan108 commented on boards/nxp/rd_rw612_bga/init.c:
```diff
@@ -0,0 +1,44 @@
+#include <fsl_clock.h>
+#include <fsl_power.h>
+#include "fsl_io_mux.h"
+
```
I don't see a lot of iomux related API's used in this file. Probably `fsl_common.h` is what you need?

EmilioCBen replied:
I think your right, I will update the PR.

mmahadevan108 commented:
@EmilioCBen , https://github.com/zephyrproject-rtos/zephyr/pull/87594 has merged. Please help fix merge conflicts.

mmahadevan108 commented:
@anangl , can you help review this PR.

decsny commented:
it's only nxp changes, assignee was wrong
